File: ZenCare.Model/Responses/RegisterResponse.cs
namespace ZenCare.Model.Responses;

public class RegisterResponse
{
    public int UserId { get; set; }
    public int ClientProfileId { get; set; }
    public string Username { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Role { get; set; } = string.Empty;
    public bool IsActive { get; set; }
    public DateTime CreatedAt { get; set; }
    public string Message { get; set; } = string.Empty;
}

File: ZenCare.Mobile/lib/models/register_response.dart
class RegisterResponse {
  RegisterResponse({
    required this.userId,
    required this.clientProfileId,
    required this.username,
    required this.email,
    required this.role,
    required this.isActive,
    required this.createdAt,
    required this.message,
  });

  final int userId;
  final int clientProfileId;
  final String username;
  final String email;
  final String role;
  final bool isActive;
  final DateTime createdAt;
  final String message;

  factory RegisterResponse.fromJson(Map<String, dynamic> json) {
    return RegisterResponse(
      userId: json['userId'] as int,
      clientProfileId: json['clientProfileId'] as int,
      username: json['username'] as String? ?? '',
      email: json['email'] as String? ?? '',
      role: json['role'] as String? ?? '',
      isActive: json['isActive'] as bool? ?? true,
      createdAt: DateTime.parse(json['createdAt'] as String),
      message: json['message'] as String? ?? 'Account created successfully. Please sign in.',
    );
  }
}
